src/App.tsx

```diff
@@ -1,204 +1,84 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useState } from 'react';
+import { UserWarning } from './Components/UserWarning/UserWarning';
+import { TodoList } from './Components/TodoList';
+import { Footer } from './Components/Footer';
+import { ErrorMessage } from './Components/ErrorMessage';
+import { Todo } from './types/Todo';
+import { getTodos, getTodosByStatus } from './api/todos';
+import { Errors } from './types/Errors';
+import { Status } from './types/Status';
+import { Header } from './Components/Header';
+import { USER_ID } from './utils/userId';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
-
-  return (
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
+  const [editedTodo] = useState<Todo | null>(null);
+  const [updateProcessing] = useState(false);
+  const [error, setError] = useState<Errors | null>(null);
+  const [filterStatus, setFilterStatus] = useState<Status>(Status.All);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setError(Errors.LoadError));
+
+    switch (filterStatus) {
+      case Status.All:
+        getTodos(USER_ID)
+          .then(setVisibleTodos)
+          .catch(() => setError(Errors.LoadError));
+        break;
+
+      case Status.Active:
+        getTodosByStatus(USER_ID, false)
+          .then(setVisibleTodos)
+          .catch(() => setError(Errors.LoadError));
+        break;
+
+      case Status.Completed:
+        getTodosByStatus(USER_ID, true)
+          .then(setVisibleTodos)
+          .catch(() => setError(Errors.LoadError));
+        break;
+
+      default:
+        break;
+    }
```

Comment:
```suggestion
```
looks like redundant to fetch todos each time when changed filterStatus. and every time will be fethc all todos and then also byStatus. 
fetch ont time all todos and just filter them when changing filterStatus